tests/test_domain_py.py

```diff
@@ -1351,3 +1352,16 @@ def test_warn_missing_reference(app, status, warning):
     assert 'index.rst:6: WARNING: undefined label: no-label' in warning.getvalue()
     assert ('index.rst:6: WARNING: Failed to create a cross reference. A title or caption not found: existing-label'
             in warning.getvalue())
+
+
+@pytest.mark.sphinx(confoverrides={'nitpicky': True})
+@pytest.mark.parametrize('include_options', (True, False))
+def test_signature_line_number(app, include_options):
+    text = (".. py:function:: foo(bar : string)\n" +
+            ("   :noindexentry:\n" if include_options else ""))
+    doc = restructuredtext.parse(app, text)
+    xrefs = list(doc.traverse(condition=addnodes.pending_xref))
```

Comment:
This should be `.findall()`

A

Reply:
Thanks --- fixed.  I had thought that `traverse` already returned an iterator, but I see that `traverse(...)` is equivalent to `list(findall(...))`